remoto/connection.py

```diff
@@ -21,9 +21,16 @@ def __init__(self, hostname, logger=None, sudo=False, threads=1, eager=True,
         self.interpreter = interpreter or 'python%s' % sys.version_info[0]
 
         if eager:
-            if detect_sudo:
-                self.sudo = self._detect_sudo()
-            self.gateway = self._make_gateway(hostname)
+            try:
+                if detect_sudo:
+                    self.sudo = self._detect_sudo()
+                self.gateway = self._make_gateway(hostname)
+            except OSError:
```

Comment:
Both can raise, because they both call `execnet.makegateway`, so it's whichever is called first, depending on the `if`. I don't think doing this twice would be any better. And you probably don't want modifications to *execnet*.